examples/apps/echo/client/echo_client.py

```diff
@@ -164,36 +167,55 @@ def Main(args=None):
 	requester_id = secrets.token_hex(32)
 	session_iv = utility.generate_iv()
 	session_key = utility.generate_key()
-	encrypted_session_key = utility.generate_encrypted_key(
-		session_key, worker_obj.encryption_key)
-	requester_nonce = secrets.token_hex(32)
+	requester_nonce = secrets.token_hex(16)
 	# Create work order
-	wo_submit_json = jrpc_request.WorkOrderSubmitJson(3, 6000, "JSON-RPC",
+	wo_submit_json = jrpc_request.WorkOrderSubmitJson(3,
 		work_order_id, worker_id, workload_id, requester_id,
-		encrypted_session_key, session_iv, requester_nonce,
-		worker_encryption_key=base64.b64decode(
-			worker_obj.encryption_key).hex(),
+		session_key, session_iv, requester_nonce,
+		result_uri="http://localhost:9834", notify_uri="http://localhost:9764",
+		worker_encryption_key=worker_obj.encryption_key,
 		data_encryption_algorithm="AES-GCM-256")
 	wo_submit_json.add_in_data(message)
 
 	# Sign work order
 	private_key = utility.generate_signing_keys()
-
+	wo_submit_json.add_encrypted_request_hash()
+	wo_submit_json.add_requester_signature(private_key)
 	# Submit work order
-	direct_wrapper.init_work_order(config)
-	direct_wrapper.work_order_submit(wo_submit_json, encrypted_session_key, 
-		worker_obj, private_key, session_key, session_iv)
 
-	# Retrieve result
-	wo_get_result_json = jrpc_request.WorkOrderGetResultJson(4, work_order_id)
-	direct_wrapper.work_order_get_result(
-		wo_get_result_json, session_key, session_iv)
+	direct_wrapper.create_work_order_adaptor(
+		config).work_order_submit(
+			wo_submit_json.get_params(),
+			wo_submit_json.get_in_data(),
+			wo_submit_json.get_out_data(),
+			id=45
+		)
 
+	# Retrieve result
+	res = direct_wrapper.create_work_order_adaptor(
+		config).work_order_get_result(
+			work_order_id,
+			id=54
+		)
+	while("result" not in res):
+		res = direct_wrapper.create_work_order_adaptor(
+			config).work_order_get_result(
+				work_order_id,
+				id=57
+			)
+		if "result" not in res:
+			logger.info(" Received Response : %s, \n \n ", res)
+			time.sleep(3)
+
+	utility.decrypted_response(
+			json.dumps(res), session_key, session_iv)
 	# Retrieve receipt
-	wo_receipt_retrieve_json = \
-		jrpc_request.WorkOrderReceiptRetrieveJson(5, work_order_id)
-	direct_wrapper.init_work_order_receipt(config)
-	direct_wrapper.work_order_receipt_retrieve(wo_receipt_retrieve_json)
+
+	direct_wrapper.create_work_order_receipt_adaptor(
+		config).work_order_receipt_retrieve(
```

Comment:
work_order_receipt_retrieve() function does not return any result ?

Reply:
yes it is